tagstudio/src/qt/widgets/preview_panel.py

```diff
@@ -546,7 +588,17 @@ def update_widgets(self):
                         ratio,
                         update_on_ratio_change=True,
                     )
-                    self.file_label.setText("\u200b".join(str(filepath)))
+                    file_str: str = ""
+                    sep_color: str = "#777777"  # Gray
+                    for i, part in enumerate(filepath.parts):
+                        part_ = part.strip(os.path.sep)
+                        if i == 0:
+                            file_str += f"{"\u200b".join(part_)}<a style='color: {sep_color}'><b>{os.path.sep}</a></b>"
+                        elif i != 0 and i != len(filepath.parts) - 1:
+                            file_str += f"{"\u200b".join(part_)}<a style='color: {sep_color}'><b>{os.path.sep}</a></b>"
```

Comment:
in front of the loop

```python
separator = f"<a style='color: {sep_color}'><b>{os.path.sep}</b></a>"
```


```suggestion
                            file_str += f"{"\u200b".join(part_)}{separator}"
                        elif i != 0 and i != len(filepath.parts) - 1:
                            file_str += f"{"\u200b".join(part_)}{separator}"
```

Reply:
actually no, both these blocks are identical, so they can be merged together instead of having two different conditions.

Reply:
I was messing with the three-case structure that I didn't even realized when two of the cases had converged... thanks for pointing this out 🙃👍